src/arrayMethodIndexOf.js

```diff
@@ -7,7 +7,29 @@
  */
 function applyCustomIndexOf() {
   [].__proto__.indexOf2 = function(searchElement, fromIndex) {
-    // write code here
+    if (!arguments.length) {
+      return -1;
+    }
+
+    let i;
+
+    if (fromIndex > 0) {
+      i = fromIndex;
+    } else if (fromIndex < 0 && Math.abs(fromIndex) < this.length) {
+      i = this.length + fromIndex;
+    } else {
+      i = 0;
+    }
+
+    for (i; i < this.length; i++) {
+      if ((this[i] === searchElement)
+        || (Number.isNaN(this[i])
+        && Number.isNaN(searchElement))) {
```

Comment:
the last bracket should be placed on the new line as well

Reply:
ok